model/src/payouts/quanto.rs

```diff
@@ -0,0 +1,609 @@
+
+        let long = self
+            .initial_margin_long()?
+            .to_signed()
+            .context("Could not convert long's initial margin to bitcoin::SignedAmount")?;
+        let long = long + fee_offset_long + pnl.long();
+        let long = long.to_unsigned().unwrap_or(Amount::ZERO);
+
+        let short = self
+            .initial_margin_short()?
+            .to_signed()
+            .context("Could not convert short's initial margin to bitcoin::SignedAmount")?;
+        let short = short + fee_offset_short + pnl.short();
+        let short = short.to_unsigned().unwrap_or(Amount::ZERO);
+
+        Ok(Payout {
+            interval,
+            long,
+            short,
+        })
+    }
+
+    /// Price interval at which the party going long gets liquidated i.e. their payout amount equals
+    /// zero.
+    fn long_liquidation_interval(&self) -> Result<RangeInclusive<u64>> {
+        let initial_margin = self
+            .initial_margin_long()
+            .context("Could not calculate long's initial margin")?;
+        let initial_margin = initial_margin
+            .to_signed()
+            .context("Could not convert long's initial margin to bitcoin::SignedAmount")?;
+        let effective_initial_margin =
+            initial_margin + self.fee_offset.as_signed_amount(Position::Long);
+        let effective_initial_margin = effective_initial_margin
+            .to_unsigned()
+            .context("Could not convert long's effective initial margin to bitcoin::Amount")?;
+
+        let thresholds = LiquidationThresholds::new(
+            effective_initial_margin,
+            self.n_contracts,
+            self.initial_price,
+            self.multiplier,
+        )
+        .context("Could not calculate liquidation thresholds")?;
+
+        Ok(0..=thresholds.long())
+    }
+
+    /// Price interval at which the party going short gets liquidated i.e. their payout amount
+    /// equals zero.
+    fn short_liquidation_interval(&self) -> Result<RangeInclusive<u64>> {
+        let initial_margin = self
+            .initial_margin_short()
+            .context("Could not calculate short's initial margin")?;
+        let initial_margin = initial_margin
+            .to_signed()
+            .context("Could not convert short's initial margin to bitcoin::SignedAmount")?;
+        let effective_initial_margin =
+            initial_margin + self.fee_offset.as_signed_amount(Position::Short);
+        let effective_initial_margin = effective_initial_margin
+            .to_unsigned()
+            .context("Could not convert short's effective initial margin to bitcoin::Amount")?;
+
+        let thresholds = LiquidationThresholds::new(
+            effective_initial_margin,
+            self.n_contracts,
+            self.initial_price,
+            self.multiplier,
+        )
+        .context("Could not calculate liquidation thresholds")?;
+
+        Ok(thresholds.short()..=maia_core::interval::MAX_PRICE_DEC)
```

Comment:
👀 In `maia_core`'s documentation we talk about `MAX_PRICE_DEC` being tied to `BTCUSD`. But in reality it all depends on the maximum value that the oracle attests to.

I think we can safely assume that it will still be using 20 digits for ETHUSD. But eventually this should depend on the contract symbol.